### DISCUSSION TITLE
App remounting on language change

### DISCUSSION THREAD

**Discussion by HrNanev** [i18next/next-i18next#1796] [Q&A] [answered]
Hello,

In our application we have several pages with forms. We use react-final-form to manage the forms. It keeps some state internally - the values in the fields, whether or not they are valid, touched etc. I believe this is achieved through react context and providers.

However, if the user changes the language, the whole app is unmounted and remounted, causing all state to be lost. Not just the state in react-final-form, but also any other react state set inside the pages.

As discussed here https://github.com/isaachinman/next-i18next/issues/1769#issuecomment-1096443759 this is intentional.

In theory, we could wrap appWithTranslation in another HOC and keep our state there, and then pass it down to our forms. However, this is quite inconvenient.

So my question is why is this remounting necessary and is there a way to avoid it?

Thanks!

**Comment by isaachinman** [comment, marked as answer]
Please see my answer here:

https://github.com/isaachinman/next-i18next/issues/1769#issuecomment-1158553082
